internal/engine/compiler/cpuid_amd64.go

```diff
@@ -0,0 +1,43 @@
+package compiler
+
+const (
+	cpuFeatureSSE3   = uint64(1)
+	cpuFeatureSSE4_1 = uint64(1) << 19
+	cpuFeatureSSE4_2 = uint64(1) << 20
+)
+
+const (
+	cpuExtraFeatureABM = uint64(1) << 5 // Advanced Bit Manipulation
+)
+
+type cpuFeatureFlags struct {
+	flags      uint64
+	extraFlags uint64
+}
+
+func loadCpuFeatureFlags() cpuFeatureFlags {
+	return cpuFeatureFlags{
+		flags:      cpuidAsBitmap(1, 0),
+		extraFlags: cpuidAsBitmap(0x80000001, 0),
+	}
+}
+
+// cpuid exposes the CPUID instruction to the Go layer (https://www.amd.com/system/files/TechDocs/25481.pdf)
+// implemented in impl_amd64.s
+func cpuid(arg1, arg2 uint32) (eax, ebx, ecx, edx uint32)
+
+// cpuidAsBitmap combines the result of invoking cpuid to uint64 bitmap
+func cpuidAsBitmap(arg1, arg2 uint32) uint64 {
+	_ /* eax */, _ /* ebx */, ecx, edx := cpuid(1, 0)
```

Comment:
Shouldn't this be `cpuid(arg1, arg2)`?

Reply:
For what it's worth, tests pass with this change, on all my computers.

Reply:
🙈 likely

Reply:
confirmed on my old laptop too